### PR TITLE
Removed unnecessary computation and fixed residual in control gravity costs

### PR DESCRIPTION
**Pull request by cmastalli**
This PR handles the comments that I left in https://github.com/loco-3d/crocoddyl/pull/888#pullrequestreview-588392158 plus a few extra improvements of both cost functions: control gravity and control gravity contact. Concretely, these is the list of actions:
 1. Remove unnecessary computation regarding the actuation model.
 2. Adapted the unittest and cost / contact numdiff routines for properly testing 1.
 3. Deprecated constructors as they don't need to pass the actuation model.
 4. Fixed the residual computation (i.e., r = tau - g(q) rather than r = g(q) - tau)

Let me give you explanations for each point. First, it is not needed to pass the actuation model as the actuation data is stored by the action models as designed choice (point 1). Second, I believe you guys got confused as the unittest (concretely,  the numdiff routines) were not written to update actuation data, and as a consequence, you introduced point 1 (point 2). Third, I have to deprecated all the constructor functions that pass the actuation model as this not anymore needed (point 3). Finally, I noticed that the residual vectors were computed with opposite sign. However, I also saw that this was not compatible with the code and PR documentation (see  https://github.com/loco-3d/crocoddyl/pull/888#issue-528146304).

@edantec and @proyan -- this PR is important for you. Pay particular attention to point 4.

### CODE REVIEW

**Review by proyan — APPROVED** [review]
Thanks for the fix Carlos,
I just have one comment about the Deprecation of the constructor with actuation_model. You are right the actuation model does not have a place in the constructor. I think we can remove it completely from the code, and don't have to bother with the deprecation. 
Other than that, this PR is okay to merge

File: include/crocoddyl/multibody/costs/control-gravity-contact.hxx
Comment (cmastalli) [reply include/crocoddyl/multibody/costs/control-gravity-contact.hxx]: Indeed, I did want to deprecate the constructors but I need your consensus for that!

File: include/crocoddyl/multibody/costs/control-gravity-contact.hxx
Comment (cmastalli) [reply include/crocoddyl/multibody/costs/control-gravity-contact.hxx]: done!

I have removed the deprecated code.

**Comment by cmastalli** [comment]
@proyan - thanks for the code review. I will wait for the CI check before merging it. Have a nice weekend!

**Pull request merged**
